app/controllers/application_controller.rb

```diff
@@ -24,6 +24,65 @@ def render_404
     render file: Rails.root.join("public", "404"), layout: false, status: "404"
   end
 
+    def put_and_get_search_params_in_session(controller_name, params_hash , filter)
```

Comment:
Assignment Branch Condition size for put_and_get_search_params_in_session is too high. [97.35/15]<br>Line is too long. [83/80]<br>Inconsistent indentation detected.<br>Space found before comma.